mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -216,16 +217,24 @@ impl State for ValidateTakerPayment {
         };
 
         let confirmations = min(watcher_ctx.data.taker_payment_confirmations, TAKER_SWAP_CONFIRMATIONS);
+        let confirm_taker_payment_input = ConfirmPaymentInput {
+            payment_tx: taker_payment_hex.clone(),
+            secret_hash: watcher_ctx.data.secret_hash.clone(),
+            // Todo: waiting for confirmation of evm swap payment state initialization should be checked by watchers too
```

Comment:
@caglaryucekaya I added this todo related to EVM watchers

Reply:
Watchers data doesn't have the `swap_contract_address` yet to pass to `wait_for_confirmations` which will lead to skipping the check of payment state initialization, which might lead to failing of payment validation in some cases. I think watchers shouldn't use RPCs that get data from multiple servers anyways which cause these race conditions. I just added this as a note for you since ETH swap watchers are still WIP.

Reply:
Watchers first validate if the `swap_contract_address` in the payment matches either the `swap_contract_address` or `fallback_swap_contract` in its contract. Afterwards they use the address in the payment for the rest of the process. I'll handle that part for the `wait_for_confirmations` in my next PR.